src/seedu/addressbook/Main.java

```diff
@@ -78,17 +78,22 @@ private void exit() {
         System.exit(0);
     }
 
-    /** Reads the user command and executes it, until the user issues the exit command.  */
+    /** Reads the user command and executes it, until the user issues the exit command or
+     *  a storage operation exception occurs.
+     */
     private void runCommandLoopUntilExitCommand() {
         Command command;
-        do {
-            String userCommandText = ui.getUserCommand();
-            command = new Parser().parseCommand(userCommandText);
-            CommandResult result = executeCommand(command);
-            recordResult(result);
-            ui.showResultToUser(result);
-
-        } while (!ExitCommand.isExit(command));
+        try {
+            do {
+                String userCommandText = ui.getUserCommand();
+                command = new Parser().parseCommand(userCommandText);
+                CommandResult result = executeCommand(command);
+                recordResult(result);
+                ui.showResultToUser(result);
+            } while (!ExitCommand.isExit(command));
+        } catch (StorageOperationException e) {
+            ui.showToUser(e.getMessage());
```

Comment:
Instead of using the default exception message, you can have a more customised message for the user.